cmd/world/root/root_test.go

```diff
@@ -134,19 +142,94 @@ func TestCreateStartStopPurge(t *testing.T) {
 	assert.Assert(t, healthResponse.IsServerRunning)
 	assert.Assert(t, healthResponse.IsGameLoopRunning)
 
+	// Restart cardinal
+	rootCmd.SetArgs([]string{"cardinal", "restart", "--detach"})
+	err = rootCmd.Execute()
+	assert.NilError(t, err)
+
+	// Check cardinal health after restart
+	resp, err = http.Get("http://127.0.0.1:3333/health")
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, 200)
+	err = json.NewDecoder(resp.Body).Decode(&healthResponse)
+	assert.NilError(t, err)
+	assert.Assert(t, healthResponse.IsServerRunning)
+	assert.Assert(t, healthResponse.IsGameLoopRunning)
+
 	// Stop cardinal
 	rootCmd.SetArgs([]string{"cardinal", "stop"})
 	err = rootCmd.Execute()
 	assert.NilError(t, err)
 
-	// Check cardinal health
+	// Check cardinal health (expected error)
 	_, err = http.Get("http://127.0.0.1:3333/health")
 	assert.Error(t, err,
 		"Get \"http://127.0.0.1:3333/health\": dial tcp 127.0.0.1:3333: connect: connection refused")
+}
 
-	// Purge cardinal
-	rootCmd.SetArgs([]string{"cardinal", "purge"})
-	err = rootCmd.Execute()
+func TestDev(t *testing.T) {
+	// Create Cardinal
+	gameDir, err := os.MkdirTemp("", "game-template")
+	os.Chdir(gameDir)
+	assert.NilError(t, err)
+
+	//set tea ouput to variable
+	teaOut := &bytes.Buffer{}
+	createCmd := getCreateCmd(teaOut)
+	createCmd.SetArgs([]string{gameDir})
+
+	err = createCmd.Execute()
 	assert.NilError(t, err)
 
+	defer func() {
+		// Remove dir
+		os.RemoveAll(gameDir)
+	}()
+
+	// Start cardinal dev
+	rootCmd.SetArgs([]string{"cardinal", "dev"})
+	go func() {
+		err := rootCmd.Execute()
+		assert.NilError(t, err)
+	}()
+
+	// Check cardinal health for 300 second, waiting to download dependencies
+	isServerRunning := false
+	isGameLoopRunning := false
+	count := 0
```

Comment:
nit:I think this would be clearer:
```
timeout := time.Now().Add(300*time.Second)
for !(isServerRunning && isGameLoopRunning) && time.Now().Before(timeout) {
...
```

Reply:
updated